src/MatrixClientPeg.js

```diff
@@ -40,6 +40,14 @@ function deviceId() {
 class MatrixClientPeg {
     constructor() {
         this.matrixClient = null;
+
+        // These are the default options used when Lifecycle.js
+        // starts the client. These can be altered when the
+        // 'will_start_client' event is dispatched.
+        this.opts = {
+            pendingEventOrdering: "detached",
```

Comment:
I think the react sdk relies on this being the case, so I think this should be always be set when we construct a MatrixClient, rather than being in a mutable options object.